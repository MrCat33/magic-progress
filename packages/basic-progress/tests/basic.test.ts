import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import glob from 'fast-glob'
import { readFile } from 'fs/promises'

describe('Vue', () => {
  describe('fixtures', async () => {
    const root = resolve(__dirname, '..')
    const files = await glob('tests/fixtures/*.{vue,js,ts}', {
      cwd: root,
      onlyFiles: true,
    })

    for (const file of files) {
      it(file.replace(/\\/g, '/'), async () => {
        const filepath = resolve(root, file)
        // const code = transfrom(await readFile(filepath,'utf-8'))
        expect(filepath).toMatchSnapshot()
      })
    }
  })
})
