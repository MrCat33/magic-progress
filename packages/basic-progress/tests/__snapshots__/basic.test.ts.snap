// Vitest Snapshot v1

exports[`Vue > fixtures > tests/fixtures/basic.vue 1`] = `
"<script setup lang=\\"ts\\">
   console.log('test util')
</script>

<template>
  hello world
</template>
"
`;
